src/apps/beefy/positions.test.ts

```diff
@@ -0,0 +1,47 @@
+import { BaseBeefyVault } from './api'
+import { getDailyYieldRatePercentage } from './positions'
+
+const apyBreakdownWithCorrectComponents = {
+  vaultApr: 0.1,
+  clmApr: 0.2,
+  tradingApr: 0.3,
+}
+
+const apyBreakdownWithIncorrectComponents = {
+  unsupportedComponent: 0.1,
+  totalApy: 0.5,
+}
+
+const apyBreakdownWithRewardPoolApr = {
+  ...apyBreakdownWithCorrectComponents,
+  rewardPoolApr: 0.4,
+}
+
+const vault: BaseBeefyVault = {
+  type: 'gov',
+  subType: 'cowcentrated',
+} as BaseBeefyVault
+
+describe('getDailyYieldRatePercentage', () => {
```

Comment:
can we add test cases for returning 0? It should be for when `undefined` and I also see a isNaN check, not sure when that can happen in practice

Reply:
added two new tests to cover these cases in [3dff1ad](https://github.com/valora-inc/hooks/pull/631/commits/3dff1ad7b2287522c5413ca6c99f54f3aa884f3e)